content/spin/v1/dynamic-configuration.md

```diff
@@ -5,65 +5,52 @@ date = "2022-03-14T00:22:56Z"
-  - [Component Custom Config](#component-custom-config)
-- [Custom Config Providers](#custom-config-providers)
-  - [Environment Variable Provider](#environment-variable-provider)
-  - [Vault Config Provider](#vault-config-provider)
-    - [Vault Config Provider Example](#vault-config-provider-example)
 - [Runtime Configuration](#runtime-configuration)
+  - [Application Variables Runtime Configuration](#application-variables-runtime-configuration)
+    - [Environment Variable Provider](#environment-variable-provider)
+    - [Vault Application Variable Provider](#vault-application-variable-provider)
+      - [Vault Application Variable Provider Example](#vault-application-variable-provider-example)
   - [Key Value Store Runtime Configuration](#key-value-store-runtime-configuration)
+    - [Redis Key Value Store Provider](#redis-key-value-store-provider)
+    - [Azure CosmosDB Key Value Store Provider](#azure-cosmosdb-key-value-store-provider)
   - [SQLite Storage Runtime Configuration](#sqlite-storage-runtime-configuration)
+    - [LibSQL Storage Provider](#libsql-storage-provider)
   - [LLM Runtime Configuration](#llm-runtime-configuration)
+    - [Remote Compute Provider](#remote-compute-provider)
 
-Spin applications may define custom configuration which can be looked up by
-component code via the [spin-config interface](https://github.com/fermyon/spin/blob/main/wit/ephemeral/spin-config.wit).
+Configuration for Spin applications using features such as [application variables](https://developer.fermyon.com/spin/variable),
+[key value storage](https://developer.fermyon.com/spin/kv-store-api-guide), [SQL storage](https://developer.fermyon.com/spin/sqlite-api-guide)
+and [Serverless AI](https://developer.fermyon.com/spin/serverless-ai-api-guide) can be provided
+dynamically via [runtime configuration](#runtime-configuration).
 
-## Custom Config Variables
-
-Application-global custom config variables are defined in the top-level `[variables]`
-section. These entries aren't accessed directly by components but are referenced
-by [component config](#component-custom-config) value templates. Each entry must
-either have a `default` value or be marked as `required = true`. "Required" entries
-must be [provided](#custom-config-providers) with a value.
-
-Configuration keys may only contain lowercase letters and underscores between letters:
-
-<!-- @nocpy -->
-
-```toml
-[variables]
-api_host = { default = "api.example.com" }
-api_key = { required = true }
-```
+## Runtime Configuration
 
-### Component Custom Config
+For many Spin features, configuration for the backing implementation can be supplied dynamically, i.e. during the application runtime,
+requiring no changes to the application code itself. We call these backing implementations 'providers' and their configuration data
+is stored in the runtime configuration file (`runtime-config.toml`). This file is then passed in via the `--runtime-config-file` flag
```

Comment:
Potential follow-up: We don't appear to have a definitive guide to the runtime configuration file and/or whole examples, just snippets per feature-specific configuration below.  Something to track for later?